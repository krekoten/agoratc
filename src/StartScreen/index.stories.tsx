import React from 'react'

import type { Meta, StoryObj } from '@storybook/react'

import StartScreen from '.'

const meta: Meta<typeof StartScreen> = {
  component: StartScreen,
}

export default meta

type Story = StoryObj<typeof StartScreen>

export const Layout: Story = {
  render: () => <StartScreen />,
};
