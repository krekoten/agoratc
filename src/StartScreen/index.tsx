import React from "react"

import { Box, Button } from "@mui/material"

interface StartParams {
  onStart: () => void
}

function StartScreen({ onStart }: StartParams): JSX.Element {
  return <Box sx={{display: "flex", flexDirection: "column", alignItems: "center", margin: 2}}>
    <Button className="start" variant="contained" onClick={onStart}>Start the Quiz</Button>
  </Box>
}

export default StartScreen
