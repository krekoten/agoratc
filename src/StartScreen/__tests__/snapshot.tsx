import React from "react"
import renderer from "react-test-renderer"

import StartScreen from ".."

test("StartScreen Snapshot", () => {
  const tree = renderer.create(<StartScreen onStart={() => {}} />).toJSON()
  expect(tree).toMatchSnapshot()
})
