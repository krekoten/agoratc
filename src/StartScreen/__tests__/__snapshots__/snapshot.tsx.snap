// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartScreen Snapshot 1`] = `
<div
  className="MuiBox-root css-ytygsf"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium start css-sghohy-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Start the Quiz
  </button>
</div>
`;
